packages/babel-parser/test/helpers/run-fixture-tests.js

```diff
@@ -0,0 +1,189 @@
+        );
+      }
+    }
+  }
+}
+
+function runAutogeneratedParseTests(
+  parse,
+  prefix,
+  task,
+  onlyCompareErrors = false,
+) {
+  const { expect, options } = task;
+  const testFn = task.disabled ? it.skip : it;
+
+  const expected = deserialize(expect.loc, options, expect.code);
+  const title = `${prefix}/${task.title}`;
+  const toStartPosition = ({ startLine = 1, startColumn = 0 }) =>
+    `(${startLine}, ${startColumn})`;
+
+  toFuzzedOptions(options)
+    .map(([adjust, options], index) => ({
+      ...task,
+      title: `${title} start = ${toStartPosition(options)}`,
+      adjust,
+      options,
+      expected,
+      filename: task.actual.loc,
+      source: task.actual.code,
+      original: index === 0,
+    }))
+    .forEach(test =>
+      testFn(test.title, () => runParseTest(parse, test, onlyCompareErrors)),
+    );
+}
+
+const toJustErrors = result => ({
+  threw: result.threw,
+  ast: result.ast && { errors: result.ast.errors },
+});
+
+function runParseTest(parse, test, onlyCompareErrors) {
+  const { adjust, expected, source, filename, options } = test;
+
+  if (expected.threw && expected.ast) {
+    throw Error(
+      "File expected.json exists although options specify throws. Remove expected.json.",
+    );
+  }
+
+  const actual = parseWithRecovery(parse, source, filename, options);
+  const difference = new Difference(
+    adjust,
+    onlyCompareErrors ? toJustErrors(expected) : expected,
+    onlyCompareErrors ? toJustErrors(actual) : actual,
+  );
+
+  // No differences means we passed and there's nothing left to do.
+  if (difference === Difference.None) return;
+
+  const error = FixtureError.fromDifference(difference, actual);
+
+  // If we're not overwriting the current values with whatever we get this time
+  // around, then we have a legitimate error that we need to report.
+  if (CI || !OVERWRITE) throw error;
+
+  // We only write the output of the original test, not all it's auto-generated
+  // variations.
+  if (!test.original) return;
+
+  const testLocation = test.taskDir;
+
+  // FIXME: We're just maintaining the legacy behavior of storing *just* the
+  // error `message` here, which differs from the error's `toString()` that we
+  // store for each error in the `errors` array. In both cases, we should
+  // serialize the full error to be able to property test locations,
+  // reasonCodes, etc.
+  const throws = !!actual.threw && actual.threw.message;
+  const optionsLocation = join(testLocation, "options.json");
+
+  // We want to throw away the contents of `throws` here.
+  // eslint-disable-next-line no-unused-vars
+  const { throws: _, ...oldOptions } = readJSON(optionsLocation);
+  const newOptions = { ...oldOptions, ...(throws && { throws }) };
+
+  // Store (or overwrite) the options file if there's anything to record,
+  // otherwise remove it.
+  if (Object.keys(newOptions).length > 0) {
+    writeFileSync(optionsLocation, stringify(newOptions, null, 2), "utf-8");
```

Comment:
To be consistent with `.editorconfig`: https://github.com/babel/babel/blob/8ccd1219f830784ead6178180f87fd23e0bf9bd9/.editorconfig#L4-L7


we should insert a trailing `\n`.

Reply:
Sure thing, just want to confirm that this is new behavior that we want, since I was following the existing code (which doesn't add a newline):

1. https://github.com/babel/babel/blob/main/packages/babel-parser/test/helpers/runFixtureTests.js#L68
2. https://github.com/babel/babel/blob/main/packages/babel-parser/test/helpers/runFixtureTests.js#L138

Super easy and happy to add, just don't want to break anything if something else is expecting it not to have newlines.

Reply:
Oh yes please, even if it means changing every fixture. It's super annoying that when I manually touch an `option.json` then my editor adds a newline which wasn't there.

Reply:
Done: https://github.com/babel/babel/pull/14201/commits/198ad6bcf2a18fc394de65506ee76d751e58919e